### PR TITLE
hdrgen: add extra space on AlignDeclaration only when needed

### PR DESCRIPTION
**Pull request by ljmf00**
This will avoid trailing whitespaces in some situations, for example, on the
described by the change of the test.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -984,7 +984,13 @@ public:
     {
         buf.writestring("align ");
         if (d.ealign)
-            buf.printf("(%s) ", d.ealign.toChars());
+        {
+            buf.printf("(%s)", d.ealign.toChars());
+            AttribDeclaration ad = cast(AttribDeclaration)d;
+            if (ad.decl && ad.decl.dim == 1)
```
Comment (RazvanN7) [inline src/dmd/hdrgen.d:990]: I think that this is going to wrongly elide the space for empty `align` declarations:

```d
align (1) {}
```

**Comment by RazvanN7** [comment]
While you are at it, can you please add header generation tests for all the variants and corner cases of the align declaration? That would include:

- `align(exp):` with and without declarations
- `align:` with and without declarations
- `align{}` - with no declarations

Thanks!

File: src/dmd/hdrgen.d (hunk 1 above)
Comment (ljmf00) [reply src/dmd/hdrgen.d]: Changed. I also fixed duplicated code introduced by #1139 .

**Comment by ljmf00** [comment]
> While you are at it, can you please add header generation tests for all the variants and corner cases of the align declaration? That would include:
> 
> * `align(exp):` with and without declarations
> * `align:` with and without declarations
> * `align{}` - with no declarations
> 
> Thanks!

Yes, added tests for the described situations. Can you check now?

**Review by RazvanN7 — APPROVED** [review]
